include/oneapi/dpl/pstl/zip_view_impl.h

```diff
@@ -0,0 +1,388 @@
+
+        iterator() = default;
+
+        constexpr iterator(iterator<!Const> i)
+            requires Const && (std::convertible_to<std::ranges::iterator_t<Views>, std::ranges::iterator_t<const Views>> && ...)
+            : current_(std::move(i.current_)) {}
+
+    private:
+        template <typename... Iterators>
+        constexpr iterator(const Iterators&... iterators)
+            : current_(iterators...) {}
+    public:
+
+        return_tuple_type operator*() const {
+
+            auto __tr = [](auto&&... __args) -> decltype(auto) { return return_tuple_type(__args...);};
+            return apply_to_tuple(__tr, [](auto it) -> decltype(auto) { return *it;}, current_);
+        }
+
+        constexpr return_tuple_type operator[]( difference_type n ) const
+            requires all_random_access<Const, Views...>
+        {
+            return *(*this + n);
+        }        
+
+        constexpr iterator& operator++() {
+            zip_view::apply_to_tuple([](auto& it) { return ++it; }, current_);
+            return *this;
+        }
+
+        constexpr void operator++(int) {
+            ++*this;
+        }
+
+        constexpr iterator operator++(int) requires all_forward<Const, Views...> {
+            auto tmp = *this;
+            ++*this;
+            return tmp;
+        }
+
+        constexpr iterator& operator--() requires all_bidirectional<Const, Views...> {
+
+            zip_view::apply_to_tuple([](auto& it) { return --it; }, current_);
+            return *this;
+        }
+
+        constexpr iterator operator--(int) requires all_bidirectional<Const, Views...> {
+            auto tmp = *this;
+            --*this;
+            return tmp;
+        }
+
+        constexpr iterator& operator+=(difference_type n)
+            requires all_random_access<Const, Views...>
+        {
+            zip_view::apply_to_tuple([n](auto& it) { return it += n; }, current_);
+            return *this;
+        }
+
+        constexpr iterator& operator-=(difference_type n)
+            requires all_random_access<Const, Views...>
+        {
+            zip_view::apply_to_tuple([n](auto& it) { return it -= n; }, current_);
+            return *this;
+        }
+
+        friend constexpr bool operator==(const iterator& x, const iterator& y)
+            requires ( std::equality_comparable<std::conditional_t<!Const, std::ranges::iterator_t<Views>,
+                                                                           std::ranges::iterator_t<const Views>>> && ... )
+        {
+            if constexpr (all_bidirectional<Const, Views...>) {
+                return x.current_ == y.current_;
+            } else {
+                return x.compare_equal(y, std::make_index_sequence<sizeof...(Views)>());
+            }
+        }
+
+        friend constexpr auto operator<=>(const iterator& x, const iterator& y)
+            requires all_random_access<Const, Views...>
+        {
+            return x.current_ == y.current_;
```

Comment:
Seems like spaceship should be used. 

```suggestion
            return x.current_ <=> y.current_;
```